pkg/testutil/prometheus.go

```diff
@@ -171,6 +184,33 @@ func (p *Prometheus) Start() error {
 	return nil
 }
 
+func (p *Prometheus) WaitPrometheusUp(ctx context.Context) error {
+	if !p.running {
+		return errors.New("method Start was not invoked.")
+	}
+	return runutil.Retry(time.Second, ctx.Done(), func() error {
+		r, err := http.Get(fmt.Sprintf("http://%s/metrics", p.addr))
```

Comment:
Might be better to check the `/-/ready` endpoint? IIRC `/metrics` is ready sooner than the instance is ready to answer queries for example.